src/quicklist.c

```diff
@@ -1829,6 +2006,68 @@ int quicklistTest(int argc, char *argv[], int accurate) {
             quicklistRelease(ql);
         }
 
+        TEST("Comprassion Plain node") {
+            quicklistisSetPackedThreshold(1);
+            quicklist *ql = quicklistNew(-2, 1);
+            for (int i = 0; i < 500; i++)
+                /* Set to 256 to allow the node to be triggered to compress,
+                 * if it is less than 48(nocompress), the test will be successful. */
+                quicklistPushHead(ql, genstr("hello", i), 256);
+
+            quicklistIter *iter = quicklistGetIterator(ql, AL_START_TAIL);
+            quicklistEntry entry;
+            int i = 0;
+            while (quicklistNext(iter, &entry)) {
+                char *h = genstr("hello", i);
+                if (strcmp((char *)entry.value, h))
+                    ERR("value [%s] didn't match [%s] at position %d",
+                        entry.value, h, i);
+                i++;
+            }
+            quicklistReleaseIterator(iter);
+            quicklistRelease(ql);
+        }
+
+
+        TEST("NEXT plain node")
+        {
+            packed_threshold = 1;
+            quicklist *ql = quicklistNew(-2, options[_i]);
+            quicklistPushHead(ql, "hello1", 6);
+            quicklistPushHead(ql, "hello2", 6);
+            quicklistPushHead(ql, "hello3", 6);
+            quicklistEntry entry;
+            quicklistIter *iter = quicklistGetIterator(ql, AL_START_TAIL);
+            quicklistNext(iter, &entry);
+            quicklistReleaseIterator(iter);
+            assert(strcmp("hello1", (char *)entry.value) == 0);
```

Comment:
This way we can verify that when compress is 1, the second entry will be correctly compressed.
Since the minimum threshold for lzf compression is 48, the second entry needs to be modified to be greater than 48.
```suggestion
            quicklistPushHead(ql, "hello1", 6);
            quicklistPushHead(ql, genstr("hello2", 64), 64);
            quicklistPushHead(ql, "hello3", 6);
            quicklistEntry entry1, entry2, entry3;
            quicklistIter *iter = quicklistGetIterator(ql, AL_START_TAIL);
            quicklistNext(iter, &entry1);
            quicklistNext(iter, &entry2);
            quicklistNext(iter, &entry3);
            quicklistReleaseIterator(iter);
            assert(strcmp("hello1", (char *)entry1.value) == 0);
            assert(strcmp("hello3", (char *)entry3.value) == 0);
            if (options[_i] == 1)
                assert(entry2.node->encoding == QUICKLIST_NODE_ENCODING_LZF);
```
```